go/ngql/types.go

```diff
@@ -22,12 +22,20 @@ type typeMap map[typeMapKey]graphql.Type
 type typeMapKey struct {
 	h             hash.Hash
 	boxedIfScalar bool
+	typeMode      graphQLTypeMode
 }
 
 func NewTypeMap() *typeMap {
 	return &typeMap{}
 }
 
+type graphQLTypeMode bool
+
+const (
+	inputMode  = graphQLTypeMode(true)
+	outputMode = graphQLTypeMode(false)
```

Comment:
More idiomatic:
```go
const (
  inputMode  = iota
  outputMode = iota
)
```
and `typeMode` as an `int`.

This would also be a logical place to explain the difference between input and output types, as you explained to me.